modules/switch/react/spec/Switch.spec.tsx

```diff
@@ -1,44 +1,182 @@
-      .at(0)
-      .getDOMNode()
-      .getAttribute('id');
+    expect(getByTestId(testIds[0]).id).not.toBe(getByTestId(testIds[1]).id);
+  });
+
+  it('should fire the onChange event handler when clicked', () => {
+    const cb = jest.fn();
+    const {getByRole} = render(<Switch checked={true} onChange={cb} />);
+
+    fireEvent.click(getByRole('checkbox'));
+    fireEvent.click(getByRole('checkbox'));
+    expect(cb).toHaveBeenCalledTimes(2);
+  });
+
+  it('should be a `type` of "checkbox"', () => {
+    const {getByRole} = render(<Switch onChange={cb} />);
+    expect(getByRole('checkbox')).toHaveProperty('type', 'checkbox');
+  });
+
+  it('should have a `tabIndex` of 0', () => {
+    const {getByRole} = render(<Switch onChange={cb} />);
+    expect(getByRole('checkbox')).toHaveProperty('tabIndex', 0);
+  });
+
+  describe('Switch styles', () => {
+    describe('When disabled', () => {
+      test('the input element should have a "not-allowed" cursor when disabled', () => {
+        const {getByRole} = render(<Switch disabled={true} onChange={cb} />);
+        expect(getByRole('checkbox')).toHaveStyleRule('cursor', 'not-allowed');
+      });
+
+      test('the background element should have a pale background color when disabled', () => {
+        const {container} = render(<Switch disabled={true} onChange={cb} />);
+        const backgroundElement = getBackground(container);
+        expect(backgroundElement).toHaveStyleRule('background-color', canvas.colors.soap400);
+      });
+    });
+
+    describe('When not disabled', () => {
+      test('the input element should have a "pointer" cursor when not disabled', () => {
+        const {getByRole} = render(<Switch onChange={cb} />);
+        expect(getByRole('checkbox')).toHaveStyleRule('cursor', 'pointer');
+      });
+
+      test('the background element should have a non-neutral background when not disabled and checked', () => {
+        const {container} = render(<Switch checked onChange={cb} />);
+        const backgroundElement = getBackground(container);
+        expect(backgroundElement).toHaveStyleRule('background-color', canvas.colors.blueberry500);
+      });
+
+      test('the background element should have a neutral background color when not disabled and unhecked', () => {
+        const {container} = render(<Switch checked={false} onChange={cb} />);
+        const backgroundElement = getBackground(container);
+        expect(backgroundElement).toHaveStyleRule('background-color', canvas.colors.licorice200);
+      });
+
+      test('the circle element should be to the very left (`translateX(0)`) when unchecked', () => {
+        const {container} = render(<Switch checked={false} disabled={true} onChange={cb} />);
+        const circleElement = getCircle(container);
+        expect(circleElement).toHaveStyleRule('transform', 'translateX(0)');
+      });
+
+      test('the circle element should be to the very right (`translateX(spacing.s)`) when unchecked', () => {
```

Comment:
Test like this are hard to reason about when they fail. What does it mean if the `transform` rule is different? What if we accomplish the same visual output using different styling rules? Is that still okay?

Reply:
Yup, you're right, the specific style rule should be removed here